From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Fri, 15 Aug 2026 00:00:00 +0000
Subject: [PATCH] libhybris-egl-server: take QOpenGLTexture from QtOpenGL

The compositor half of the libhybris server-buffer integration does not build:

    hardwareintegration/compositor/libhybris-egl-server/
    libhybriseglserverbufferintegration.cpp:8:10: fatal error:
    QtGui/QOpenGLTexture: No such file or directory

QOpenGLTexture moved from QtGui to QtOpenGL in Qt 6.0. Every other source under
hardwareintegration/compositor - drm-egl-server, dmabuf-server,
shm-emulation-server, vulkan-server, linux-dmabuf-v1, wayland-egl and
wayland-eglstream-controller - already includes <QtOpenGL/QOpenGLTexture>. This
one was missed because it only builds when libhybris is present, which upstream
CI does not have.

Only the include is wrong here: unlike the client-side plugin in qtbase, this
CMakeLists already lists both libhybris-egl-server-buffer.xml and
server-buffer-extension.xml, and the compositor plugins pick up Qt::OpenGL
through Qt::WaylandCompositorPrivate.

Upstream-Status: Pending
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
---
 .../compositor/libhybris-egl-server/libhybriseglserverbufferintegration.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/hardwareintegration/compositor/libhybris-egl-server/libhybriseglserverbufferintegration.cpp
+++ b/src/hardwareintegration/compositor/libhybris-egl-server/libhybriseglserverbufferintegration.cpp
@@ -5,7 +5,7 @@
 #include "libhybriseglserverbufferintegration.h"
 
 #include <QtGui/QOpenGLContext>
-#include <QtGui/QOpenGLTexture>
+#include <QtOpenGL/QOpenGLTexture>
 #include <hybris/eglplatformcommon/hybris_nativebufferext.h>
 #include <wayland-server-core.h>
 
